### PR TITLE
DS18B20 - CRC check added to lua module, minor optimization; change of ow module example

### PR DESCRIPTION
**Pull request by vsky279**
No issue connected with this PR.

Make sure all boxes are checked (add x inside the brackets) when you submit your contribution, remove this sentence before doing so.

- [X] This PR is for the `dev` branch rather than for `master`.
- [X] This PR is compliant with the [other contributing guidelines](https://github.com/nodemcu/nodemcu-firmware/blob/dev/CONTRIBUTING.md) as well (if not, please describe why).
- [X] I have thoroughly tested my contribution.
- [X] The code changes are reflected in the documentation at `docs/*`.

I have 7 DS18B20 devices connected to 1 bus and I start to receive some wrong readouts. This may be connected to an hw issue. The module does not check CRC for readouts so wrong temperatures are reported. CRC check was added.

Beside this some minor optimization of the module.

This PR also changes nasty example in `ow` module documentation. The example was using `tmr.delay(1000000)`...

### CODE REVIEW

File: docs/modules/ow.md [hunk 1]
```diff
@@ -145,48 +145,42 @@ Issues a 1-Wire rom select command. Make sure you do the `ow.reset(pin)` first.
...
+-- 18b20 Example
+pin = 3
 ow.setup(pin)
-count = 0
-repeat
-  count = count + 1
-  addr = ow.reset_search(pin)
-  addr = ow.search(pin)
-  tmr.wdclr()
-until (addr ~= nil) or (count > 100)
```
Comment (vsky279) [inline docs/modules/ow.md]: No point of cycling through all sensors in this example. Only first found sensor is read.

**Comment by HHHartmann** [comment]
Thanks for this contribution. 
The build currently fails or new luacheck With the following messages:

`   Checking lua_modules/ds18b20/ds18b20.lua          2 warnings
362
363    lua_modules/ds18b20/ds18b20.lua:88:76: line contains trailing whitespace
364    lua_modules/ds18b20/ds18b20.lua:102:43: line contains trailing whitespace   `

**Comment by vsky279** [comment]
It's great we have such check. Corrected.

**Comment by marcelstoer** [comment]
Also clean-up a nasty `ow` module example.

**Pull request merged**